### ISSUE TITLE
gameoverlayrenderer.so cannot be preloaded & steam.sh and dota.sh SEGFAULTs

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/steam-for-linux#3365]
After i woke up my desktop from a suspend state, i fired up Steam to play some DotA. I started it, and right before the DotA logo disappears and usually it would greet me the store page, it crashed. I checked up, and 

```
ERROR: ld.so: object '/home/psycho/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
Uploading dump (out-of-process) [proxy '']
/tmp/dumps/crash_20140621122845_1.dmp
/home/psycho/.local/share/Steam/SteamApps/common/dota 2 beta/dota.sh: line 77:  6724 Segmentation fault      (core dumped) ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
Game removed: AppID 570 "Dota 2", ProcID 6724 
Finished uploading minidump (out-of-process): success = yes
response: CrashID=bp-76b8594a-42d5-42b3-b78b-93a382140621
```

this happened and DotA crashed. I tried it a matter of times, everytime crashing. I even unchecked the `Enable Steam Overlay In-Game' option and it still crashed with the same error. I restarted Steam so i could tell my friend about this, but whenever i tried to open the Friends window, it crashed with

```
ExecSteamURL: "steam://open/friends"
Uploading dump (out-of-process) [proxy '']
/tmp/dumps/assert_20140621122644_1.dmp
/home/psycho/.local/share/Steam/steam.sh: line 755:  6602 Segmentation fault      (core dumped) $STEAM_DEBUGGER "$STEAMROOT/$PLATFORM/$STEAMEXE" "$@"
Finished uploading minidump (out-of-process): success = yes
response: CrashID=bp-fa7fbc6f-c624-4590-9a27-d16712140621
```

EDIT: Steam also crashes when i try to open the Store

```
Generating new string page texture 555: 256x256, total string texture memory is 393.22 KB
Uploading dump (out-of-process) [proxy '']
/tmp/dumps/assert_20140621123407_1.dmp
/home/psycho/.local/share/Steam/steam.sh: line 755:  6886 Segmentation fault      (core dumped) $STEAM_DEBUGGER "$STEAMROOT/$PLATFORM/$STEAMEXE" "$@"
Finished uploading minidump (out-of-process): success = yes
response: CrashID=bp-8c2e3119-d648-424d-a7e8-699842140621
```

TL;DR: Steam crashes when trying to open the Friends window, DotA crashed because of gameoverlayrenderer.so. (Both dota.sh and steam.sh SEGFAULTs)

Recent Failure Reports:

```
    Sat Jun 21 09:12:29 2014 GMT: file ''/tmp/dumps/crash_20140621121218_29.dmp'', upload yes: ''CrashID=bp-e1d8cd6d-1ee9-4e78-8795-0b3242140621''
    Sat Jun 21 09:14:26 2014 GMT: file ''/tmp/dumps/crash_20140621121251_1.dmp'', upload yes: ''CrashID=bp-307e977b-d2c9-4f63-b135-f18942140621''
    Sat Jun 21 09:14:56 2014 GMT: file ''/tmp/dumps/crash_20140621121355_1.dmp'', upload yes: ''CrashID=bp-33f4008f-a917-4fd6-9101-8e8482140621''
    Sat Jun 21 09:19:28 2014 GMT: file ''/tmp/dumps/assert_20140621121647_1.dmp'', upload yes: ''CrashID=bp-1392d218-8406-4d5b-83c2-214a52140621''
    Sat Jun 21 09:20:51 2014 GMT: file ''/tmp/dumps/crash_20140621121713_1.dmp'', upload no: ''Couldn't connect to server''
    Sat Jun 21 09:21:08 2014 GMT: file ''/tmp/dumps/assert_20140621121935_1.dmp'', upload yes: ''CrashID=bp-ee0d6c8b-1f26-4615-99dc-ed1e12140621''
    Sat Jun 21 09:21:18 2014 GMT: file ''/tmp/dumps/assert_20140621122108_3.dmp'', upload yes: ''CrashID=bp-d2a5db9d-ede6-46ee-9fa0-76ea72140621''
    Sat Jun 21 09:21:30 2014 GMT: file ''/tmp/dumps/assert_20140621122108_5.dmp'', upload yes: ''CrashID=bp-e33fa85e-ea63-44b8-b7ef-0e3c52140621''
    Sat Jun 21 09:30:09 2014 GMT: file ''/tmp/dumps/crash_20140621122845_1.dmp'', upload yes: ''CrashID=bp-76b8594a-42d5-42b3-b78b-93a382140621''
    Sat Jun 21 09:30:31 2014 GMT: file ''/tmp/dumps/crash_20140621122730_1.dmp'', upload no: ''Couldn't connect to server''
    Sat Jun 21 09:33:53 2014 GMT: file ''/tmp/dumps/assert_20140621122644_1.dmp'', upload yes: ''CrashID=bp-fa7fbc6f-c624-4590-9a27-d16712140621''
```

All of them are from these issues.
